verification-right.md

```diff
@@ -0,0 +1,139 @@
+   * Is the data valid? Is the attestation on-chain and not revoked? See the simple [Verification](verification).
+
+OK, let's see this in action.
+
+## As the <span class="label-role verifier">verifier</span>: create a challenge
+
+To generate a random, unique piece of data, we'll use a package called [uuid].
+Install it:
+
+```bash
+# with yarn
+yarn add uuid
+# or with npm
+npm install uuid
+```
+
+Create a new file `challenge.js`, and paste the following code into it:
+
+```javascript
+import { v4 as uuid } from 'uuid'
+const challenge = uuid()
+console.log('challenge: ', challenge)
+```
+
+Run the code by running this command in your terminal, still within your `kilt-rocks` directory:
+
+```bash
+node challenge.js
+```
+
+You should see in your logs the `uuid`, that will be used as a challenge; it should look like something like this: `9dbbad48-aad0-4280-aeca-4c4072c82625`.
+
+Copy it, you'll need it in the next step.
+
+## As the <span class="label-role claimer">claimer</span>: solve the challenge and prepare the data
+
+Let's put together the data you would send back to the <span class="label-role verifier">verifier</span>, as the <span class="label-role claimer">claimer</span>.
+
+Create a new file `claim-with-proof.js`.
+
+Paste the following code into it (make sure to replace `<challenge>` and `<attestedClaimJSON>` with the data you copied earlier):
+
+```javascript
+const Kilt = require('@kiltprotocol/sdk-js')
+
+// <challenge> = challenge received from the verifier = uuid you copied from above
+const challenge = <challenge>
+
+// <claimer mnemonic> = claimer mnemonic generated in the Identity step
+const claimer = Kilt.Identity.buildFromMnemonic(<claimer mnemonic>)
+// sign the challenge as the claimer with your private identity
+const proof = claimer.signStr(challenge)
+
+// same data as in to the simple "Verification" step
+const attestedClaimStruct = JSON.parse(JSON.stringify(<attestedClaimJSON>));
+
+const dataToVerify = {
+  proof: proof,
+  attestedClaimStruct
+}
+
+console.log('dataToVerifyJSON: ', JSON.stringify(dataToVerify))
+```
+
+Run the code by running this command in your terminal, still within your `kilt-rocks` directory:
+
+```bash
+node claim-with-proof.js
+```
+
+You should see in your logs the `dataToVerify`.
+
+Copy it, you'll need it in the next step.
+
+## As the <span class="label-role verifier">verifier</span>: verify the `proof` and `attestedClaim`
+
+Create a new file `verification-with-challenge.js`.
+
+Paste the following code into it (make sure to replace `<dataToVerify>` and `<challenge>` with the relevant objects):
+
+```javascript
+const Kilt = require('@kiltprotocol/sdk-js')
+
+const { proof, attestedClaimStruct } = JSON.parse(<dataToVerifyJSON>)
```

Comment:
everywhere else we use `JSON.parse(JSON.stringify(...))` except here? Is there a reason for that?

Reply:
Uh, these `JSON.parse(JSON.stringify...` didn't make sense - cleaned up.